src/ConnectedMode/Migration/MSBuildFileProvider.cs

```diff
@@ -31,18 +38,111 @@ namespace SonarLint.VisualStudio.ConnectedMode.Migration
     [PartCreationPolicy(CreationPolicy.NonShared)]
     internal class MSBuildFileProvider : IFileProvider
     {
+        private readonly IServiceProvider serviceProvider;
         private readonly ILogger logger;
+        private readonly IThreadHandling threadHandling;
+        private readonly IFileSystem fileSystem;
+
+        internal static readonly string[] FileSearchPatterns = new string[] {
+            "*.ruleset",
+            "*.props",
+            "*.targets",
+            "*.csproj",
+            "*.vbproj"
+        };
+
+        internal static readonly string[] ExcludedDirectores = new string[]
+        {
+            "\\bin\\",
+            "\\obj\\"
+        };
```

Comment:
Testing this on SLVS itself was picking up lots of generated .props and .targets files in the `obj` directory. I've added simple filtering to exclude both `obj` and `bin` directories.